### ISSUE TITLE
Invoke certificate-update event when a valid certificate is renewed

### ISSUE THREAD

**Issue by gsanchietti** [NethServer/dev#5174]
The `certificate-update` event must be fired every time a valid certificate is renewed.
This applies to:

- Let's Encrypt certificates

Also take care that `certificate-update` doesn't regenerated the built-in certificate: see also http://dev.nethserver.org/issues/3435

Affected packages:
- nethserver-base
- nethserver-letsencrypt

**Comment by gsanchietti** [comment]
In nethserver-testing:
- nethserver-letsencrypt-1.1.2-1.2.gbf1d159.ns7.noarch.rpm
- nethserver-base-3.0.11-1.8.gafa549a.ns7.noarch.rpm

*Test case 1*
- Open the "Server certificate" page and make a change
- Verify `/etc/pki/tls/certs/NSRV.crt` has been updated:
```
stat /etc/pki/tls/certs/NSRV.crt
openssl x509 -noout -dates -in /etc/pki/tls/certs/NSRV.crt
```

*Test case 2*
- After test case 1, execute the certificate-update event: 
```
signal-event certificate-update
```
- Verify `/etc/pki/tls/certs/NSRV.crt` hasn't been changed

*Test case 3*
- Configure a server with Let's Encrypt and request a new certificate
- Verify the `certificate-update` has been executed

*Test case 4*
- After test case 3, force the renewal of Let's Encrypt certificate
- Verify the `certificate-update` has been executed

**Comment by gsanchietti** [comment]
Released in 7.3.1611/base:
- nethserver-ndpi-1.1.0-1.ns7.noarch.rpm
- kmod-xt_ndpi-2.0.0-1.ns7.x86_64.rpm

**Comment by DavidePrincipi** [comment]
Test case 1 & 2 were OK

**Comment by filippocarletti** [comment]
Test case 1 & 2: OK
Before:
```
notBefore=Jun 17 17:21:15 2016 GMT
notAfter=Jun 15 17:21:15 2026 GMT

```

After cert changein web UI:
```
notBefore=Dec 15 11:04:49 2016 GMT
notAfter=Dec 13 11:04:49 2026 GMT

```
Calling certificate-update doesn't modifiy the cert:
```
# signal-event certificate-update 
# stat /etc/pki/tls/certs/NSRV.crt
Modify: 2016-12-15 12:04:49.596714989 +0100

```

Test case 3: OK
Obtained a new LE cert:
`Dec 15 12:09:00 demo esmith::event[3334]: Event: certificate-update SUCCESS`

Test case 4: FAILED
Details of LE cert: 
```
# openssl x509 -noout -dates -in /etc/letsencrypt/live/demo.nethesis.it/cert.pem 
notBefore=Dec 15 10:20:00 2016 GMT
notAfter=Mar 15 10:20:00 2017 GMT

```
Request and obtain a new cert:
`/usr/libexec/nethserver/letsencrypt-certs -f -d demo.nethesis.it`

Confirm it's new:
```
# openssl x509 -noout -dates -in /etc/letsencrypt/live/demo.nethesis.it/cert.pem 
notBefore=Dec 15 10:24:00 2016 GMT
notAfter=Mar 15 10:24:00 2017 GMT

```

certificate-update event is not fired.

**Comment by filippocarletti** [comment]
Using nethserver-base-3.0.11-1.11.ge1b9cb1.ns7.noarch test case 4 passes.
```
Dec 15 16:22:02 demosecurity esmith::event[10064]: Event: certificate-update SUCCESS
notBefore=Dec 15 14:22:00 2016 GMT
```

**Comment by gsanchietti** [comment]
Released in 7.3.1611/base
- nethserver-letsencrypt-1.1.3-1.ns7.noarch.rpm
- nethserver-base-3.0.12-1.ns7.noarch.rpm
